test/unittests/typed_value_test.cpp

```diff
@@ -0,0 +1,75 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "test/utils/typed_value.hpp"
+#include <gtest/gtest.h>
+
+using namespace fizzy;
+using namespace fizzy::test;
+
+TEST(typed_value, construct_contexpr)
+{
+    constexpr TypedValue i32{int32_t{-1}};
+    static_assert(i32.type == ValType::i32);
+    static_assert(i32.value.i64 == uint32_t(-1));
+    EXPECT_EQ(i32.type, ValType::i32);
+    EXPECT_EQ(i32.value.i64, uint32_t(-1));
```

Comment:
Why have both `sattic_assert` and `EXPECT_EQ`?

Reply:
Performing check in constexpr and non-contexpr context. This first one does not give code coverage.